moduleroot/appveyor.yml

```diff
@@ -0,0 +1,58 @@
+<%
+# Workaround to support boolean supports_windows and hash+array supported_oss
+# settings until one form is no longer used
+supports_windows = (
+  @configs['supports_windows'] == true \
+  || (
+    @configs['supported_oss'].is_a?(Hash) \
+    && @configs['supported_oss'][@configs[:puppet_module]].is_a?(Array) \
+    && @configs['supported_oss'][@configs[:puppet_module]].include?('windows')
+  )
+)
+
+if supports_windows -%>
+version: 1.1.x.{build}
+skip_commits:
+  message: /^\(?doc\)?.*/
+clone_depth: 10
+init:
+- SET
+- 'mkdir C:\ProgramData\PuppetLabs\code && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\facter && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\hiera && exit 0'
+- 'mkdir C:\ProgramData\PuppetLabs\puppet\var && exit 0'
+environment:
+  matrix:
+<% @configs['matrix'].each do |matrix| -%>
+  - PUPPET_GEM_VERSION: <%= matrix['PUPPET_GEM_VERSION'] %>
+    RUBY_VER: <%= matrix['RUBY_VER'] %>
+<% end -%>
+<% if @configs['matrix_extras'] -%>
+<%   @configs['matrix_extras'].each do |extra| -%>
+  - PUPPET_GEM_VERSION: <%= extra['PUPPET_GEM_VERSION'] %>
+    RUBY_VER: <%= extra['RUBY_VER'] %>
+<%   end -%>
+<% end -%>
+matrix:
+  fast_finish: true
+install:
+- SET PATH=C:\Ruby%RUBY_VER%\bin;%PATH%
+- <%= @configs['appveyor_bundle_install'] %>
+- type Gemfile.lock
+build: off
+test_script:
+- bundle exec puppet -V
+- ruby -v
+<% @configs['test_script'].each do |script| -%>
+- <%= script %>
+<% end -%>
+notifications:
+- provider: Email
+  to:
+  - nobody@nowhere.com
+  on_build_success: false
+  on_build_failure: false
+  on_build_status_changed: false
+<% else -%>
+# Appveyor is only enabled on modules that support windows
```

Comment:
We don't currently have the tests setup to do so, but long-term testing should cover whether facts and other native code can be plugin-synced to all (agent-)supported platforms and not break a run when none of the dependencies are installed.

For some modules, like stdlib, we should already be testing everything on windows, as it is a supported platform.

or, 5. manually set `appveyor.yml` to `unmanaged: true` in the local `.sync.yml`.

apt and stdlib are the only ones where I started playing around with appveyor.